### ISSUE TITLE
how to handle recursive mounts

### ISSUE THREAD

**Issue by lastephey** [NERSC/podman-hpc#40]
Based on some testing from @scanon  it looks like a recursive mount is needed for cvmfs

After some stracing in podman, it looks like their mounts are recursive by default: https://docs.google.com/document/d/1EkvvG2dXQX0DoWAoO_qk5kp5fjCSveCOTAiuJzSjaJs/edit?usp=sharing

Should we make all of our mounts recursive by default too? Or if not, we need to enable the mount type to be specified in the hook.

**Comment by lastephey** [comment]
I switched `bind` to `rbind` in https://github.com/NERSC/podman-hpc/blob/main/podman_hpc/hook_tool.py#L52

```
stephey@nid001001:~> cat /usr/lib/python3.6/site-packages/podman_hpc/hook_tool.py | grep bind
def bind_mount(src, tgt):
    bind mount a file into a namespace.
    subprocess.check_output(["mount", "--rbind", src, tgt])
    set up to do copies and bind mounts
    # handle the bind case
    for f in mod["bind"]:
                bind_mount(fp, tgt2)
            bind_mount(src, tgt2)
stephey@nid001001:~> 
```

I tested with basic mpi and gpu test and so far it seems ok

```
stephey@nid001001:~> srun -n 2 podman-hpc run --rm --mpi registry.nersc.gov/library/nersc/mpi4py:3.1.3 python3 -m mpi4py.bench helloworld
Hello, World! I am process 0 of 2 on nid001001.
Hello, World! I am process 1 of 2 on nid001001.
```

```
stephey@nid001001:~> podman-hpc run --rm --gpu registry.nersc.gov/library/nersc/mpi4py:3.1.3 nvidia-smi
Tue Mar 14 19:18:10 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.65.01    Driver Version: 515.65.01    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA A100-SXM...  Off  | 00000000:03:00.0 Off |                    0 |
| N/A   25C    P0    52W / 400W |      0MiB / 40960MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA A100-SXM...  Off  | 00000000:41:00.0 Off |                    0 |
| N/A   24C    P0    49W / 400W |      0MiB / 40960MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA A100-SXM...  Off  | 00000000:82:00.0 Off |                    0 |
| N/A   25C    P0    50W / 400W |      0MiB / 40960MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA A100-SXM...  Off  | 00000000:C1:00.0 Off |                    0 |
| N/A   24C    P0    51W / 400W |      0MiB / 40960MiB |      0%      Default |
|                               |                      |             Disabled |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
stephey@nid001001:~> 
```



@scanon @dfox-lbl what are your thoughts about just always using `rbind`?

**Comment by dfox-lbl** [comment]
I never use anything but rbind.

**Comment by danfulton** [comment]
Resolved by #41
